### PR TITLE
Support kwargs only calls to contour

### PR DESCRIPTION
**Pull request by oscargus**
## PR Summary

Closes #24743

Check that there is at least one element in `args` before accessing it.

## PR Checklist

**Documentation and Tests**
- [ ] Has pytest style unit tests (and `pytest` passes)
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] New plotting related features are documented with examples.

**Release Notes**
- [ ] New features are marked with a `.. versionadded::` directive in the docstring and documented in `doc/users/next_whats_new/`
- [ ] API changes are marked with a `.. versionchanged::` directive in the docstring and documented in `doc/api/next_api_changes/`
- [ ] Release notes conform with instructions in  `next_whats_new/README.rst` or `next_api_changes/README.rst`

### CODE REVIEW

File: lib/matplotlib/contour.py [hunk 1]
```diff
@@ -1386,7 +1386,7 @@ def _process_args(self, *args, corner_mask=None, algorithm=None, **kwargs):
         """
         Process args and kwargs.
         """
-        if isinstance(args[0], QuadContourSet):
+        if len(args) and isinstance(args[0], QuadContourSet):
```
Comment (chahak13) [inline lib/matplotlib/contour.py:1389]: Just `if args` should work too.

Comment (oscargus) [reply lib/matplotlib/contour.py:1389]: Yeah, you are right. I'll change it. Thanks!

**Comment by chahak13** [comment]
@oscargus this will fail because of `_contour_args` in this line

```python
x, y, z = self._contour_args(args, kwargs)
```

It'll need a check in that function too the if statements don't check non-zero `len(args)`
(Sorry, I was just about to raise a PR for this too)

**Comment by chahak13** [comment]
Something like this should work (this will error out on no positional args, will need changes)

```diff python
-       if nargs <= 2:
-           z = ma.asarray(args[0], dtype=np.float64)
-           x, y = self._initialize_x_y(z)
-           args = args[1:]
-       elif nargs <= 4:
-           x, y, z = self._check_xyz(args[:3], kwargs)
-           args = args[3:]
-       else:
-           raise _api.nargs_error(fn, takes="from 1 to 4", given=nargs)
+       if 0 < nargs <= 2:
+           z = ma.asarray(args[0], dtype=np.float64)
+           x, y = self._initialize_x_y(z)
+           args = args[1:]
+       elif 2 < nargs <= 4:
+           x, y, z = self._check_xyz(args[:3], kwargs)
+           args = args[3:]
+       else:
+           raise _api.nargs_error(fn, takes="from 1 to 4", given=nargs)
```

Though, I do have a question about allowing passing `Z` as a keyword argument or let it be a positional argument and error out if none are given? This is what the `quiver` PR also did.

**Comment by oscargus** [comment]
Yeah, I noticed that as well (although it did work when running it interactively strangely enough). I was about to start extracting X, Y, and Z from the kwargs at some stage, but as this wasn't as quick as I thought I'll move this to draft for now. Feel free to prepare a better PR.

I think you fix, checking that there is at least one argument, is good to have anyway, but in the long run I think that X, Y, and Z should be extracted if kwargs (before that check).

**Pull request closed without merging**